src/core/lib/resource_quota/memory_quota.h

```diff
@@ -327,9 +355,14 @@ class BasicMemoryQuota final
   std::atomic<intptr_t> free_bytes_{kInitialSize};
   // The total number of bytes in this quota.
   std::atomic<size_t> quota_size_{kInitialSize};
+  // Assign a unique ID to each allocator.
+  std::atomic<uint64_t> curr_allocator_id_{0};
```

Comment:
why assign an id, versus `reinterpret_cast<uintptr_t>(allocator)`?

doing so would let you eliminate the id storage and passing around everywhere

Reply:
changed to hash_set of allocator ptrs.